projects/humans/skin_animated_humans/libraries/bvh_util/src/quaternion_private.h

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright 1996-2021 Cyberbotics Ltd.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef QUATERNION_PRIVATE_H
```

Comment:
Why naming this file with a private suffix? It seems it provides a public interface...
The same applies to the vector3 header file.

Reply:
It is private because it is not in the `include` folder of the library.

Reply:
I know, but still I believe it shouldn't have this private suffix because we expose a public interface here, even if that is within the bvh_util library, e.g., we do not expose private/protected methods to friend/child classes.

Reply:
For me it is exactly the same to keep or not the private suffix.
But this class is only meant for private use within the library implementation and it uses the same private suffix exactly as we also do in the C controllers folder (see for example `device_private.h`, etc). So it seems to be more coherent to keep it.
If this file would be meant for public use, then this header should be moved in the `include` folder.

Reply:
"public use" doesn't mean "use by the users of the API", but "use by other classes". The `device_private.h` and other private files should contain declarations for either friend or derived classes, not user classes.

Reply:
Note that in the src/controllers/c folder we have both public and private header files, although none of them is used by API users.